operators/pkg/controller/common/certificates/ca_reconcile.go

```diff
@@ -0,0 +1,213 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package certificates
+
+import (
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"time"
+
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/common/name"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/common/reconciler"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/utils/k8s"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/rand"
+)
+
+// CAType is a type of CA
+type CAType string
+
+const (
+	// TransportCAType is the CA used for ES transport certificates
+	TransportCAType CAType = "transport"
```

Comment:
We want the CAs used to have some distinct identifier. Defining them all in the same place makes it easy to see and discover. Since it’s just type identifiers and we don’t need plugability here, it seems like a reasonable tradeoff.